### PR TITLE
[PRTL-2575] enhance biotree’s expand/collapse functionality

### PR DESCRIPTION
**Pull request by caravinci**
## Environment to be used in testing

- [x] `prod`
- [x] `dev-oicr`
- [ ] `qa` (which version?)

## Description of Changes
Corrects the button's behaviour

### CODE REVIEW

File: src/packages/@ncigdc/modern_components/BiospecimenCard/BioTree/index.js [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+      emitterToken.remove();
+      setTreeStatus({
+        payload: {
+          expanded,
+          mounted: false,
+        },
+        type: 'TOTAL_NODES',
+      });
+    },
+    // shouldComponentUpdate({
```
Comment (samrichca) [inline src/packages/@ncigdc/modern_components/BiospecimenCard/BioTree/index.js]: can this be deleted

Comment (caravinci) [reply src/packages/@ncigdc/modern_components/BiospecimenCard/BioTree/index.js]: I've only included it as I had planned on adding enhancements on it later, but you're right... I'll add it when/if the time comes 👍

File: src/packages/@ncigdc/modern_components/BiospecimenCard/reducers.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
+export default [
```
Comment (anncatton) [inline src/packages/@ncigdc/modern_components/BiospecimenCard/reducers.js]: can we move this reducer to the `/dux` folder with the other reducers?

File: src/packages/@ncigdc/modern_components/BiospecimenCard/reducers.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  (state, { // Reducer Function
+    payload: {
+      expanded,
+      mounted,
+    } = {},
+    type,
+  }) => {
+    switch (type) {
+      case 'COLLAPSE_NODE':
```
Comment (anncatton) [inline src/packages/@ncigdc/modern_components/BiospecimenCard/reducers.js]: this is an elegant solution!

File: package.json [hunk 4]
```diff
@@ -8,7 +8,7 @@
   },
   "homepage": "https://portal.gdc.cancer.gov/",
   "scripts": {
-    "start": "concurrently --kill-others 'npm run start-ui' 'npm run relay -- --watch'",
+    "start": "concurrently --kill-others --prefix '[{name} - {time}]' -t 'HH:mm:ss' -n webpack,graphql 'npm run start-ui' 'npm run relay -- --watch'",
```
[the remark this answers is not in the document]
Comment (caravinci) [reply package.json]: it's just something I've changed to help me track process timings, which led me to notice an issue with one of Node's dependencies, when running on some of the latest MacBook Pro. (therefor my adding `fsevents` to `optionalDependencies`).
Besides, I find having information about what specific process (e.g. webpack or relay) is the one logging any given line, is better than just having a [0] and a [1]. 

image
Now, when you say it looks odd in your command line, you mean it doesn't look like this? ☝️

[the remark this answers is not in the document]
Comment (samrichca) [reply package.json]: agreed, timestamps & process names can be shown with bash/zsh settings

File: src/packages/@ncigdc/modern_components/BiospecimenCard/BiospecimenCard.js [hunk 5]
```diff
@@ -15,353 +27,468 @@ import Input from '@ncigdc/uikit/Form/Input';
...
+        treeStatusOverride && setTreeStatus({
+          payload: {
+            expanded: treeStatusOverride === 'expanded' ||
+              (treeStatusOverride === 'query matches' && total > 1),
+          },
+          type: 'OVERRIDE_NODES',
+        });
+
+        return ({
+          // allCollapsed: total === collapsed,
```
[the remark this answers is not in the document]
Comment (caravinci) [reply src/packages/@ncigdc/modern_components/BiospecimenCard/BiospecimenCard.js]: I was foreseeing the need to separate `collapse all` from `expand all`, yet forgot to remove it. 
Thanks for pointing it out. I'll amend it 👍

File: src/packages/@ncigdc/modern_components/BiospecimenCard/reducers.js (hunk 2 above)
Comment (caravinci) [reply src/packages/@ncigdc/modern_components/BiospecimenCard/reducers.js]: It's a reducer specific to this component alone, i.e. not reusable/global.
By Component Oriented Programming principles, it should remain where it is, yet I'll move it if everyone else agrees I should.

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (anncatton) [reply package.json]: odd maybe the wrong word - it looks cluttered. i'd prefer if this was something we can customize on our own machines, as sam points out

[the remark this answers is not in the document]
Comment (caravinci) [reply package.json]: Fair enough. Amending.

File: src/packages/@ncigdc/modern_components/BiospecimenCard/reducers.js (hunk 2 above)
Comment (anncatton) [reply src/packages/@ncigdc/modern_components/BiospecimenCard/reducers.js]: i'd prefer sticking with code organization we currently have in the portal (with all the reducers in one folder), if we choose as a team to reorganize at some point we can change things up then.

**Comment by caravinci** [comment]
Updated the base branch (i.e. `release/1.23.0` to `develop`)

**Pull request merged**